From eec818ccf16a153371e80882f9033cd68f46eacc Mon Sep 17 00:00:00 2001
From: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>
Date: Mon, 7 Apr 2025 17:32:58 +0200
Subject: [PATCH] ARM: dts: aspeed: Add dts for Supermicro X11SSH

This commit adds dts file for Supermicro X11SSH platform.

Based on the:
https://github.com/Keno/openbmc-linux/commit/2443d1129e95943da1ba86768934489dca548321.patch

Upstream-Status: Inappropriate [testing]
Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>
---
 arch/arm/boot/dts/aspeed/Makefile             |   1 +
 .../aspeed/aspeed-bmc-supermicro-x11ssh.dts   | 143 ++++++++++++++++++
 2 files changed, 144 insertions(+)
 create mode 100644 arch/arm/boot/dts/aspeed/aspeed-bmc-supermicro-x11ssh.dts

diff --git a/arch/arm/boot/dts/aspeed/Makefile b/arch/arm/boot/dts/aspeed/Makefile
index 220210c318fd..96e1abfba5b5 100644
--- a/arch/arm/boot/dts/aspeed/Makefile
+++ b/arch/arm/boot/dts/aspeed/Makefile
@@ -72,6 +72,7 @@ dtb-$(CONFIG_ARCH_ASPEED) += \
 	aspeed-bmc-quanta-q71l.dtb \
 	aspeed-bmc-quanta-s6q.dtb \
 	aspeed-bmc-supermicro-x11spi.dtb \
+	aspeed-bmc-supermicro-x11ssh.dtb \
 	aspeed-bmc-inventec-starscream.dtb \
 	aspeed-bmc-inventec-transformers.dtb \
 	aspeed-bmc-tyan-s7106.dtb \
diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-supermicro-x11ssh.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-supermicro-x11ssh.dts
new file mode 100644
index 000000000000..ce15f815b164
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-supermicro-x11ssh.dts
@@ -0,0 +1,143 @@
+//Based on the: https://github.com/Keno/openbmc-linux/commit/2443d1129e95943da1ba86768934489dca548321.patch
+/dts-v1/;
+
+#include "aspeed-g4.dtsi"
+#include <dt-bindings/gpio/aspeed-gpio.h>
+
+/ {
+	model = "Supermicro X11SSH BMC";
+	compatible = "supermicro,x1ssh-bmc", "aspeed,ast2400";
+
+	aliases {
+		serial4 = &uart5;
+	};
+
+	chosen {
+		stdout-path = &uart5;
+		bootargs = "console=ttyS4,115200 earlycon";
+	};
+
+	memory {
+		/* 128 MiB DDR3 memory */
+		reg = <0x40000000 0x08000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		vga_memory: framebuffer@47000000 {
+			no-map;
+			reg = <0x47000000 0x01000000>; /* 16MB */
+		};
+
+		flash_memory: region@98000000 {
+			no-map;
+			reg = <0x98000000 0x01000000>; /* 16MB */
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		uid {
+			gpios = <&gpio ASPEED_GPIO(D, 6) GPIO_ACTIVE_HIGH>;
+		};
+
+		heartbeat {
+			gpios = <&gpio ASPEED_GPIO(D, 7) GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "heartbeat";
+		};
+	};
+};
+
+&fmc {
+	status = "okay";
+	flash@0 {
+		status = "okay";
+		m25p,fast-read;
+#include "openbmc-flash-layout.dtsi"
+	};
+};
+
+&spi {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi1debug_default>;
+
+	flash@0 {
+		status = "okay";
+		label = "pnor";
+		m25p,fast-read;
+	};
+};
+
+&pinctrl {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flbusy_default &pinctrl_flwp_default
+
+			&pinctrl_vgahs_default &pinctrl_vgavs_default
+			&pinctrl_ddcclk_default &pinctrl_ddcdat_default>;
+};
+
+&mac0 {
+	status = "okay";
+
+	use-ncsi;
+	no-hw-checksum;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii1_default>;
+};
+
+
+&uart5 {
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+};
+
+&i2c2 {
+	status = "okay";
+};
+
+&i2c3 {
+	status = "okay";
+};
+
+&i2c4 {
+	status = "okay";
+};
+
+&i2c5 {
+	status = "okay";
+};
+
+&i2c6 {
+	status = "okay";
+};
+
+&i2c7 {
+	status = "okay";
+};
+
+&vuart {
+	status = "okay";
+};
+
+&lpc_ctrl {
+	status = "okay";
+	memory-region = <&flash_memory>;
+	flash = <&spi>;
+};
+
+&wdt2 {
+	status = "okay";
+};
--
2.49.0

